core/server/src/main/java/alluxio/master/journal/JournalWriter.java

```diff
@@ -379,12 +423,32 @@ public synchronized void flush() throws IOException {
           LOG.info("Rotating log file. size: {} maxSize: {}", mDataOutputStream.size(),
               mMaxLogSize);
         }
-        // rotate the current log.
-        mDataOutputStream.close();
-        completeCurrentLog();
-        mRawOutputStream = openCurrentLog();
-        mDataOutputStream = new DataOutputStream(mRawOutputStream);
+        rotateLog();
```

Comment:
Since we are using the flag approach, does it make sense to just set the flag here and only call `rotateLog` in one place?